scripts/bdc/run_dbgap_xml_gen_fallback.py

```diff
@@ -0,0 +1,471 @@
+#!/usr/bin/env python3
+"""
+Script for downloading dbGaP data dictionaries with XML generation fallback.
+Handles cases where studies are not found on dbGaP by gracefully falling back 
+to XML generation via run_xml_generator.
+"""
+
+import os
+import sys
+import shutil
+import logging
+import pandas as pd
+import argparse
+from datetime import datetime
+from get_dbgap_data_dicts import download_dbgap_study
+# Import the XML generator functionality directly
+from xml_generator import run_xml_generation
+from xml_utils import (
+    generate_xml_for_study_safe,
+    setup_logging_safe
+)
+
+# Configuration with defaults (can be overridden via command line)
+DEFAULT_CONFIG = {
+    "GEN3_CSV": "/path/to/gen3_studies_filtered.csv",
+    "PICSURE_CSV": "/path/to/pic_sure_data.csv",
+    "OUTPUT_DIR": "output",
+    "ACCESSION_FIELD": "Accession",
+    "STUDY_ID_FIELD": "study_id",
+    "STUDY_NAME_FIELD": "Study Name",
+}
+
+def read_study_Gen3(gen3_metadata_file):
+    try:
+        logging.info(f"Reading metadata file: {gen3_metadata_file}")
+        df = pd.read_csv(gen3_metadata_file)
+        df['study_id'] = df['Accession'].str.split('.').str[0]
+        logging.info(f"Found {len(df)} studies in metadata file")
+        return df.set_index('study_id')
+    except Exception as e:
+        logging.error(f"Error reading study metadata file: {e}")
+        raise
+
+def create_output_directory(base_dir):
+    os.makedirs(base_dir, exist_ok=True)
+    return base_dir
+
+def setup_config():
```

Comment:
I would move this slightly earlier to the file so it's right next to the DEFAULT_CONFIG constant that it's related to, but that might just be my preference :)